minimalkv/fsspecstore.py

```diff
@@ -122,40 +125,60 @@ def iter_keys(self, prefix: str = "") -> Iterator[str]:
         IOError
             If there was an error accessing the store.
         """
-        # List files
-        all_files_and_dirs = self._fs.find(f"{self.prefix}", prefix=prefix)
+        # `find` only lists files below a directory, so we have to split
+        # the two prefixes accordingly.
+        full_prefix = f"{self._prefix}{prefix}"
+        # Find last slash in full prefix
+        last_slash = full_prefix.rfind("/")
+        if last_slash == -1:
+            # No slash in full prefix
+            dir_prefix = ""
+            file_prefix = full_prefix
+        else:
+            dir_prefix = full_prefix[: last_slash + 1]
+            file_prefix = full_prefix[last_slash + 1 :]
+        all_files_and_dirs = self._fs.find(dir_prefix, prefix=file_prefix)
```

Comment:
can you add some explanation on what's happening here? I'm a bit confused about why there are two prefixes.

Reply:
```
        # `find` only lists files below a directory, so we have to join
        # and then split the two prefixes accordingly.
        # Example for a Boto3Store:
        # bucketname/prefix1prefix2
        # ========== -------        Boto3Store perspective
        # bucket.name self.prefix
        # ==================------- FSSpectStore perspective
        #    self._prefix    prefix
        # ===========-------------- What we want
        # dir_prefix  file_prefix
```

Added this, does that make sense?

Reply:
I'm still a bit confused. Maybe we can talk about this in person today?